### PR TITLE
fix `glooctl add route --dry-run` to not create route tables or virtual services

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2715

### CODE REVIEW

File: pkg/utils/selectionutils/route_table.go [hunk 1]
```diff
@@ -69,10 +70,6 @@ func (s *routeTableSelector) create(ctx context.Context, ref *core.ResourceRef)
 		routeTable.Metadata.Name = "default"
 	}
 
-	written, err := s.client.Write(routeTable, clients.WriteOpts{Ctx: ctx})
-	if err != nil {
-		return nil, err
-	}
-	contextutils.LoggerFrom(ctx).Infow("Created new default route table", zap.Any("routeTable", routeTable))
-	return written, nil
+	contextutils.LoggerFrom(ctx).Infow("Created new default route table object", zap.Any("routeTable", routeTable))
```
Comment (kdorosh) [inline pkg/utils/selectionutils/route_table.go:73]: let's move these log lines to the dry-run encapsulated logic where we actually create the object in the cluster

File: pkg/utils/selectionutils/virtual_service_test.go [hunk 2]
```diff
@@ -179,19 +169,6 @@ var _ = Describe("SelectorTest", func() {
...
-					List().
-					Return([]string{}, nil)
-				vsClient.EXPECT().
-					Write(getDefault(podNamespace, "default"), clients.WriteOpts{Ctx: context.Background()}).
-					Return(nil, testErr)
-
-				_, err := selector.SelectOrCreateVirtualService(context.Background(), nil)
-				Expect(err).To(HaveOccurred())
-				Expect(err).To(Equal(testErr))
-			})
```
Comment (kdorosh) [inline pkg/utils/selectionutils/virtual_service_test.go:194]: we shouldn't be deleting tests

Comment (ashleywang1) [reply pkg/utils/selectionutils/virtual_service_test.go:194]: This was due to deleting the Write call in the selectionutils code. We changed the functionality of that call, so this test is no longer testing anything.

Comment (kdorosh) [reply pkg/utils/selectionutils/virtual_service_test.go:194]: Yeah i wrote this comment before i realized where the real write was coming from, this lgtm

File: pkg/utils/selectionutils/route_table.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func (s *routeTableSelector) SelectOrCreateRouteTable(ctx context.Context, ref *
 		return found, nil
 	}
 
+	// Create a new default route table object
 	return s.create(ctx, ref)
```
Comment (kdorosh) [inline pkg/utils/selectionutils/route_table.go:56]: rename this to `build`

**Pull request merged**